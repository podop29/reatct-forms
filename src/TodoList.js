import {useState} from 'react'
import Todo from './todo'
import NewTodoForm from './NewTodoForm'
import {v4 as uuid} from "uuid"
import "./todoList.css"


function TodoList() {
    const INITIAL_VALUES = [
        {
        id:uuid(),
        todo: "feed Dog"
        },
        {
            id:uuid(),
            todo: "brush cat"
        }
    ]
    

    const [todos, setTodos] = useState(INITIAL_VALUES)

    //Removes a todo by id
    const removeTodo = (id) =>{
        const todosCopy = [...todos]
        setTodos(todosCopy.filter(todo=>todo.id!==id))
    }

    const createTodo = (data) =>{
        const todo = {
            id:uuid(),
            todo:data
        }
        setTodos([...todos,todo])
        
    }

   

    return (
        <><NewTodoForm createTodo={createTodo}/>
      <div className='todoList'>
          {todos.map(todo=>{return(
              <Todo 
              key={todo.id}
              id={todo.id}
              todo={todo.todo}
              removeTodo={removeTodo}
               />
          )})}
      </div>
      </>
    );
  }
  
  export default TodoList;
  