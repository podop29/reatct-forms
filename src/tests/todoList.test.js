import React from "react"
import {render} from "@testing-library/react"
import TodoList from "../TodoList"

it("renders witout crashing", ()=>{
    render(<TodoList/>)
})

it("matches snapshot", ()=>{
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot()
})