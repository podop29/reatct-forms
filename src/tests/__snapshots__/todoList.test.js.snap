// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todoList"
  >
    <form>
      <input
        name="todo"
        placeholder="Enter Task"
        type="text"
        value=""
      />
      <button>
        Add todo
      </button>
    </form>
  </div>
  <div
    class="todoList"
  >
    <div
      class="todo"
    >
      <h3>
        feed Dog
      </h3>
      <button>
        X
      </button>
    </div>
    <div
      class="todo"
    >
      <h3>
        brush cat
      </h3>
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
